scanners/helpers.test.js

```diff
@@ -0,0 +1,193 @@
+      const mockScanStatusResponse = require("./__testFiles__/mockCascadingScanStatusResponse.json");
+      const mockListNamespacedCustomObjectResponse = require("./__testFiles__/mockCascadingListNamespacedCustomObject.json");
+
+      mockK8sCRDApi.createNamespacedCustomObject.mockResolvedValue(
+        mockScanCreationResponse
+      );
+      mockK8sCRDApi.getNamespacedCustomObjectStatus.mockResolvedValue(
+        mockScanStatusResponse
+      );
+
+      mockK8sCRDApi.listNamespacedCustomObject.mockResolvedValue(
+        mockListNamespacedCustomObjectResponse
+      );
+
+      const findings = await cascadingScan(
+        "nmap-dummy-ssh",
+        "nmap",
+        ["-Pn", "-sV", "dummy-ssh.demo-targets.svc"],
+        {
+          nameCascade: "ncrack-ssh",
+          matchLabels: {
+            "securecodebox.io/invasive": "invasive",
+            "securecodebox.io/intensive": "high",
+          },
+        },
+        180,
+        mockK8sCRDApi,
+        mockK8sBatchApi,
+        mockPodsApi
+      );
+
+      expect(findings).toBeDefined();
+      expect(findings).toMatchSnapshot();
+    });
+    it("should throw an error if the scan fails", async () => {
+      const mockScanCreationResponse = require("./__testFiles__/mockCascadingScanCreationResponse.json");
+      const mockScanStatusResponse_Errored = require("./__testFiles__/mockCascadingScanStatusResponse_Errored.json");
+
+      const mockListNamespacedJobResponse = require("./__testFiles__/mockListNamespacedJobResponse.json");
+
+      const mockListNamespacedPodResponse = require("./__testFiles__/mockListNamespacedPodResponse.json");
+      const mockReadNamespacedPodLogResponse = require("./__testFiles__/mockReadNamespacedPodLogResponse.json");
+
+      mockK8sCRDApi.createNamespacedCustomObject.mockResolvedValue(
+        mockScanCreationResponse
+      );
+      mockK8sCRDApi.getNamespacedCustomObjectStatus.mockResolvedValue(
+        mockScanStatusResponse_Errored
+      );
+      mockK8sBatchApi.listNamespacedJob.mockResolvedValue(
+        mockListNamespacedJobResponse
+      );
+      mockPodsApi.listNamespacedPod.mockResolvedValue(
+        mockListNamespacedPodResponse
+      );
+      mockPodsApi.readNamespacedPodLog.mockResolvedValue(
+        mockReadNamespacedPodLogResponse
+      );
+
+      try {
+        await cascadingScan(
+          "nmap-dummy-ssh",
+          "nmap",
+          ["-Pn", "-sV", "dummy-ssh.demo-targets.svc"],
+          {
+            nameCascade: "ncrack-ssh",
+            matchLabels: {
+              "securecodebox.io/invasive": "invasive",
+              "securecodebox.io/intensive": "high",
+            },
+          },
+          180,
+          mockK8sCRDApi,
+          mockK8sBatchApi,
+          mockPodsApi
+        );
+      } catch (error) {
+        expect(error).toMatchInlineSnapshot(
+          `[Error: Initial Scan failed with description "Mocked Error"]`
+        );
+      }
```

Comment:
these kinds of assertions are a bit risky as the test would pass when there is not exception thrown.

there a two ways to avoid this in jest. you can either tell jest how many assertions to expect and fail if you have fewer.

or

(i prefer that way :) ): Use expect.rejects: https://jestjs.io/docs/expect#rejects
```js
test('rejects to octopus', async () => {
  await expect(Promise.reject(new Error('octopus'))).rejects.toThrow('octopus');
});
```